File: algo/src/main/java/com/hartwig/actin/algo/evaluation/pregnancy/CanUseAdequateAnticonception.java
package com.hartwig.actin.algo.evaluation.pregnancy;

import com.hartwig.actin.PatientRecord;
import com.hartwig.actin.algo.datamodel.Evaluation;
import com.hartwig.actin.algo.evaluation.EvaluationFunction;

import org.jetbrains.annotations.NotNull;

public class CanUseAdequateAnticonception implements EvaluationFunction {

    CanUseAdequateAnticonception() {
    }

    @NotNull
    @Override
    public Evaluation evaluate(@NotNull final PatientRecord record) {
        return Evaluation.NOT_EVALUATED;
    }
}

File: algo/src/main/kotlin/com/hartwig/actin/algo/evaluation/reproduction/CanUseAdequateAnticonception.kt
package com.hartwig.actin.algo.evaluation.reproduction

import com.hartwig.actin.algo.evaluation.EvaluationFactory
import com.hartwig.actin.algo.evaluation.EvaluationFunction
import com.hartwig.actin.datamodel.PatientRecord
import com.hartwig.actin.datamodel.algo.Evaluation

class CanUseAdequateAnticonception: EvaluationFunction {
    override fun evaluate(record: PatientRecord): Evaluation {
        return EvaluationFactory.notEvaluated(
            "Assumed that patient will adhere to relevant anticonception prescriptions",
            "Assumed adherence to anticonception prescriptions"
        )
    }
}